File: arrays/2sum/2sum.ts
function twoSum(nums: number[], target: number): number[] {
  const map: Record<number, number> = {};

  for (let i = 0; i < nums.length; i++) {
    if (map[target - nums[i]] !== undefined) {
      return [map[target - nums[i]], i];
    }
    map[nums[i]] = i;
  }
  return [];
}

File: arrays/2sum/2sum.go
func twoSum(nums []int, target int) []int {
	hash := map[int]int{}

	for index, num := range nums {
		if val, ok := hash[target-num]; ok {
			return []int{index, val}
		}

		hash[num] = index

	}

	return []int{}
}
